indra/newview/pipeline.cpp

```diff
@@ -7035,13 +7035,8 @@ void LLPipeline::generateExposure(LLRenderTarget* src, LLRenderTarget* dst, bool
 
         if (probe_ambiance > 0.f)
         {
-            F32 hdr_scale = sqrtf(LLEnvironment::instance().getCurrentSky()->getGamma()) * 2.f;
-
-            if (hdr_scale > 1.f)
-            {
-                exp_min = 1.f / hdr_scale;
-                exp_max = hdr_scale;
-            }
+            exp_min = dynamic_exposure_ev_offset - dynamic_exposure_ev_min;
+            exp_max = dynamic_exposure_ev_offset + dynamic_exposure_ev_max;
```

Comment:
I learned the hard way that you need to actually call the cached controls for them to work reliably. Maybe it's only an issue for bool values? Or maybe there's something wrong with the `LLCachedControl` API...

Reply:
I'm curious to hear more about this.  if the implicit casting `operator const T&()` for LLCachedControl is not functioning properly, we should figure out how to fix it

Reply:
To clarify, I would try to read the LLCachedControl bool for the first time before opening a floater to decide what UI to display, and it wouldn't work the first time.

Relevant PR:

https://github.com/secondlife/viewer/pull/2492